lib/elixir/lib/calendar/iso.ex

```diff
@@ -102,6 +102,19 @@ defmodule Calendar.ISO do
       ]
     end
 
+  [match_ordinal_date, guard_ordinal_date, read_ordinal_date] =
+    quote do
+      [
+        <<y1, y2, y3, y4, ?-, d1, d2, d3>>,
+        y1 >= ?0 and y1 <= ?9 and y2 >= ?0 and y2 <= ?9 and y3 >= ?0 and y3 <= ?9 and y4 >= ?0 and
+          y4 <= ?9 and d1 >= ?0 and d1 <= ?9 and d2 >= ?0 and d2 <= ?9 and d3 >= ?0 and d3 <= ?9,
```

Comment:
How about using making it short with `in`:
```suggestion
        y1 in ?0..?9 and y2 in ?0..?9 and y3 in ?0..?9 and y4 in ?0..?9 and
          d1 in ?0..?9 and d2 in ?0..?9 and d3 in ?0..?9,
```
or even:
```suggestion
        y1 in '09' and y2 in '09' and y3 in '09' and y4 in '09' and
          d1 in '09' and d2 in '09' and d3 in '09',
```

Reply:
Interestingly, the second form `?5 in '09' doesn't seem to work for me, but I do like the former.

There might be a performance implication here though--is list inclusion faster or slower than two equality checks? These are guards and compile-time lists so I assume either is awfully fast.

Reply:
We originally wrote in this format because it skips an `is_integer` check compared to ranges. But it is likely that the is_integer check is optimized away today. @michalmuskala would probably know. :)

Reply:
@christhekeele Ah, I see … as said I wrote it by hand … Should be:
```elixir
iex> ?5 in '0123456789'
true
```
In that case `0..9` range is shorter.